### ISSUE TITLE
rknn model accuracy severe drop

### ISSUE THREAD

**Issue by sparshgarg23** [airockchip/rknn-toolkit2#573]
So i was working on a custom resnet18 model,i performed INT8 quantization on it and exported it as an onnx. 
On the onnx model eval acc is  fine and then i converted it to rknn using below script

The model basically reads frames from video does mediapipe face detection and then performs classification using resnet18
```
from rknn.api import RKNN

rknn = RKNN(verbose=True)

# 1. Target configuration
print('--> Configuring RKNN model...')
rknn.config(
    target_platform='rk3576',
    optimization_level=2
)

# 2. Load the INT8 ONNX model with fixed input shape
print('--> Loading Quantized INT8 ONNX model...')
ret = rknn.load_onnx(
    model='/content/Gaze_Resnet_INT8.onnx',
    inputs=['input'],
    input_size_list=[[1, 3, 224, 224]]
)
if ret != 0:
    raise RuntimeError('Failed to load ONNX model!')

# 3. Build the RKNN model
print('--> Building RKNN model...')
ret = rknn.build(do_quantization=False)
if ret != 0:
    raise RuntimeError('Failed to build RKNN model!')

# 4. Export the final .rknn artifact
print('--> Exporting RKNN model...')
ret = rknn.export_rknn('Gaze_Resnet_int8.rknn')
if ret != 0:
    raise RuntimeError('Failed to export RKNN model!')

print('--> Successfully exported Gaze_Resnet_int8.rknn!')
```

now when i ran it using simulator  the model acc is the same as the onnx version but on performing the test on the actual board 3576 i found the accuracy for each frame dropped signifcantly from 80% to 27%,basically all predictions ended up being in one class.

the code for that is below in the rknn gaze inference and preprocess and other helper functions are in rknn_gaze_util

so few things on running it  the following warning message
```
the input[0] need NHWC data format, but NCHW set, the data format and data buffer will be changed to NHWC.
```
i later set inputs_pass_through=[0] in rknn.inference and removed the transpose part as well as changed how the norm_mean and norm_std are being reshaped.

on running the accuracy heavily dropped to 27% across all frames .

So i am not sure why this happened because when i ran it on colab simulator the accuracy was fine and the NHWC data format warning didnt come.

2 things come to my mind firstly 
```
in model zoo resnet and mobilenet examples rknn.config multiplies the mean and norm by 255 and during preprocessing only resizing is done.

secondly should i set do_quantixation to true and load the unquantized onnx maybe  the conversion from onnx_int8 to rknn version is affecting it.

finally in post-processing rather than  doing this 
```
def postprocess_logits(outputs, classes):
    """Applies softmax to output logits and returns label and confidence."""
    raw_logits = outputs[0][0]
    e_x = np.exp(raw_logits - np.max(raw_logits))
    probs = e_x / e_x.sum(axis=0)

    pred_idx = int(np.argmax(probs))
    return classes[pred_idx], float(probs[pred_idx]
```

its better to use the example
```
scores=softmax(output[0])
scores=np.squeeze(scores)
```
and then map those scores using classes label map
```
CLASSES = [
    "left_mirror",
    "left",
    "front",
    "center_mirror",
    "front_right",
    "right_mirror",
    "right",
    "infotainment",
    "steering_wheel",
]
```
the overall pipeloine is as follows
1.read frame
2.perform mediapipe face detection,get face bbox and use it to extract face region
3.pre-process face region crop
4.output=rknn.inference(inputs=[face_crop_preprocessed],data_format='nchw')
5.apply softmax on output and save results.
Any help will be appreciated.


[rknn_gaze_inference.py](https://github.com/user-attachments/files/32063957/rknn_gaze_inference.py)
[rknn_gaze_util.py](https://github.com/user-attachments/files/32063956/rknn_gaze_util.py)
